tweakwcs/multivariate.py

```diff
@@ -0,0 +1,112 @@
+"""
+Initial implementation of a function to fit multiple outputs 
+using astropy.modeling
+"""
+
+from astropy.modeling import FittableModel
+from astropy.modeling.models import Mapping, Const1D
+from astropy.modeling.models import math
+import numpy as np
+
+
+class _Index(FittableModel):
+
+    def __init__(self):
+        self._inputs = ()
+        self._outputs = ()
+        self._n_inputs = 1
+        self._n_outputs = 1
+        super().__init__()
+        self.inputs = ('x',)
+        self.outputs = ('x',)
+
+    @property
+    def n_inputs(self):
+        return self._n_inputs
+
+    @property
+    def n_outputs(self):
+        return self._n_outputs
+
+    def __repr__(self):
+        return f'<Index()>'
+
+    def evaluate(self, *args):
+        if len(args) != 1:
+            raise TypeError(f'Index model only takes 1 input, got {len(args)}')
+        input = args[0]
+        if np.isscalar(input):
+            return 0
+        inp_array = np.array(input)
+        if len(inp_array.shape) != 1:
+            raise ValueError('Index currently only handles 1D arrays')
+        size = inp_array.shape[0]
+        return np.arange(size)
+
+    def inverse(self):
+        raise NotImplementedError('Index model has no inverse by design')
+
+
+def _multi_output_fit(fitter, model, inputs, outputs, **kwargs):
+    """
+    Function to enable fits to models with more than one output variable.
```

Comment:
Could you, please, add some more details to the description like what does this model actually do with multiple outputs? How will the _actual_ fitting be performed? _"Function to enable fits..."_ seems too vague.